File: AquariumObject.cpp
#include "AquariumObject.h"

// Konstruktor
AquariumObject::AquariumObject() {
	X = 0;
  	Y = 0;
  	speed = 0;
}
AquariumObject::AquariumObject(float _x, float _y, double _speed) {
	X = _x;
	Y = _y;
	speed = _speed;
}
AquariumObject::~AquariumObject() {
	
}

// Getter dan setter
float AquariumObject::getX() {
	return X;
}
float AquariumObject::getY() {
	return Y;
}
float AquariumObject::getSpeed() {
	return speed;
}
void AquariumObject::setX(float _x) {
	X = _x;
}
void AquariumObject::setY(float _y) {
	Y = _y;
}
void AquariumObject::setSpeed(double _speed) {
	speed = _speed;
}
float AquariumObject::distance(AquariumObject* Obj){
  return (sqrt(fabs(pow(this->getX() - Obj->getX(), 2) + pow(this->getY() - Obj->getY(), 2))));
}

File: AquariumObject.java
/**
 * Kelas AquariumObject.
 * @author AbnerA
 */
public abstract class AquariumObject {
    /**
     * Koordinat Aquarium di (absis,ordinat).
     */
    private double absis, ordinat;
    /**
     * Kecepatan AquariumObject.
     */
    private double speed;

    /**
     * Konstruktor Dasar.
     */
    public AquariumObject() {
        this.absis = 0;
        this.ordinat = 0;
        this.speed = 0;
    }
    /**
     * Konstruktor dengan parameter.
     * @param paramAbsis absis
     * @param paramOrdinat ordinat
     * @param paramSpeed speed
     */
    public AquariumObject(final double paramAbsis, final double paramOrdinat,
            final double paramSpeed) {
        this.absis = paramAbsis;
        this.ordinat = paramOrdinat;
        this.speed = paramSpeed;
    }
    // Getter dan setter
    /**
     * Getter Absis.
     * @return nilai absis
     */
    public final double getAbsis() {
        return this.absis;
    }
    /**
     * Getter Ordinat.
     * @return nilai ordinat
     */
    public final double getOrdinat() {
        return this.ordinat;
    }
    public final double getSpeed() {
        return this.speed;
    }
    public final void setAbsis(final double paramAbsis) {
        this.absis = paramAbsis;
    }
    public final void setOrdinat(final double paramOrdinat) {
        this.ordinat = paramOrdinat;
    }
    public final void setSpeed(final double paramSpeed) {
        this.speed = paramSpeed;
    }

    //Jarak ke object lain
    public final double distance(final AquariumObject obj) {
        return (Math.sqrt(Math.abs(Math.pow((double) (
                this.getAbsis() - obj.getAbsis()),
                (double) 2)
                + Math.pow((double) (
                        this.getOrdinat() - obj.getOrdinat()), (double) 2))));
    }

    //Fungsi untuk menggerakkan objek
    public abstract void move(double time);
}
